src/org/omegat/core/threads/SearchThread.java

```diff
@@ -82,6 +84,9 @@ public void run() {
                 // bad regexp input
                 // alert user to badness
                 window.displayErrorRB(e, "ST_REGEXP_ERROR");
+            } catch (IndexOutOfBoundsException e) {
+                // replacement with group which does not exist
+                window.displayErrorRB(null, "ST_REGEXP_REPLACEGROUP_ERROR", e.getMessage().substring(e.getMessage().lastIndexOf(" ") + 1));
```

Comment:
I think this munging of the exception message is a bit dangerous. I imagine you did it this way because there's no other way to retrieve the problematic group from the exception object.

Instead of this, I'd rather detect the issue ahead of time and throw a custom exception before attempting to access the group. E.g. in `Searcher`:

```
int varId = Integer.parseInt(replaceMatcher.group(1));
if (varId > matcher.groupCount()) {
    throw new InvalidRegexGroupException(OStrings.getString("ST_REGEXP_REPLACEGROUP_ERROR", varId));
}

Reply:
One potential inconvenient of this method is that you change the interface of a public method (searchString), because it throws a new exception. Being a public method, are you sure it will not have an impact on plugins?

Reply:
If the custom exception extends `RuntimeException` then it is not a checked exception and doesn't affect the method signature.

Reply:
OK fine.
Since this exception will probably later be used in other locations, do you have any particular preference about where to put it (i.e. somewhere in org.omegat.util for example?)

Reply:
I don't know that we will necessarily use the exception elsewhere. If we do, we would probably be abstracting the entire search/replace mechanism, in which case I think an appropriate location for it would become apparent.

For now I think we can make it a nested static class within `Searcher`.

Reply:
Finally this was not necessary: since IndexOutOfBoundsException is public I could throw my own instance in the searcher, with a customized message.

Reply:
Yes, that's better than my idea. Thanks.